contracts/OXO.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.4.15;
+
+contract OXO {
+ 
+    event BidEvent(address addr, uint ammount);
+    event MarkEvent(address addr, uint x, uint y);
+    event ErrorEvent(string message);
+    event WinnerEvent(uint8 winner);
+
+    address player1;
+    address player2;
+    uint player1Bid;
+    uint player2Bid;
+    uint8[3][] grid;
+    uint8 lastMarkByPlayer = 0;
+    uint lastMarkTimestamp = 0;
+    
+    function OXO() public {
+        grid = new uint8[3][](3);
+    }
+
+    function bidAndPlay() public payable returns (bool) {
+        if (player1==0){
+            if (player2!=0){
```

Comment:
do you have any situation when player1 == 0 but player2 != 0?

Reply:
yes, when player1 decides to withdraw money before game have started

Reply:
Ow, okey)